atc/worker/gardenruntime/worker.go

```diff
@@ -23,6 +25,8 @@ import (
 	"golang.org/x/sync/errgroup"
 )
 
+var WaitingForStreamedVolumePollingInterval = 5 * time.Second
```

Comment:
Is 5 seconds too long? Some volume could be very small, especially with p2p streaming, a volume streaming could be very fast. With a 5 seconds timer, once a step needs to wait for streaming, it would take more than 5 seconds. I guess 2 seconds might be better.

Reply:
I wasn't sure since each loops does the `findVolumeForArtifact` and potentially tries to acquire the lock which could get a bit aggressive on the DB if many tasks are waiting for streaming in parallel?

Reply:
Is it possible to change to 1 or 2 seconds and try in your environment to see that would lead to more db loads?

Reply:
Skip this comment.